src/main/java/friday/logic/parser/AddCommandParser.java

```diff
@@ -42,12 +44,43 @@ public AddCommand parse(String args) throws ParseException {
         }
 
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        TelegramHandle telegramHandle = ParserUtil.parseTelegramHandle(argMultimap
-                .getValue(PREFIX_TELEGRAMHANDLE).orElse(""));
-        Consultation consultation = ParserUtil.parseConsultation(LocalDate.parse(argMultimap
-                .getValue(PREFIX_CONSULTATION).orElse("2001-01-01")));
-        MasteryCheck masteryCheck = ParserUtil.parseMasteryCheck(LocalDate.parse(argMultimap
-                .getValue(PREFIX_MASTERYCHECK).orElse("2001-01-01")));
+
+        TelegramHandle telegramHandle;
+        Consultation consultation;
+        MasteryCheck masteryCheck;
+
+        if (argMultimap.getValue(PREFIX_TELEGRAMHANDLE).isPresent()) {
+            telegramHandle = ParserUtil.parseTelegramHandle(argMultimap
+                    .getValue(PREFIX_TELEGRAMHANDLE).get());
+        } else {
+            telegramHandle = TelegramHandle.EMPTY_TELEGRAM_HANDLE;
+        }
+
+        if (argMultimap.getValue(PREFIX_CONSULTATION).isPresent()) {
+            try {
+                consultation = ParserUtil.parseConsultation(LocalDate.parse(argMultimap
+                        .getValue(PREFIX_CONSULTATION).get()));
+            } catch (DateTimeParseException e) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        Consultation.MESSAGE_CONSTRAINTS));
+            }
+        } else {
+            consultation = Consultation.EMPTY_CONSULTATION;
+        }
```

Comment:
It might be just me, but it seems abit out of place to catch the exception here. Perhaps it might be better to do LocalDate.parse within parseConsultation then catch the error there?

Reply:
Yes, I agree with you. We can change the argument of `parseConsultation` to take in a String instead.